# **Waze Project**
**Course 4 - The Power of Statistics**

# **Course 4 End-of-course project: Data exploration and hypothesis testing**

**The goal** is to apply descriptive statistics and hypothesis testing in Python to analyze the relationship between mean amount of rides and device type
<br/>

*This activity has three parts:*

**Part 1:** Imports and data loading

**Part 2:** Conduct hypothesis testing

**Part 3:** Communicate insights with stakeholders

<br/>


# **Data exploration and hypothesis testing**

### Imports and data loading




Import packages and libraries needed to compute descriptive statistics and conduct a hypothesis test.

In [3]:
import pandas as pd
from scipy import stats


In [4]:
# Load dataset into dataframe
df = pd.read_csv('waze_dataset.csv')

### Data exploration

**Note:** In the dataset, `device` is a categorical variable with the labels `iPhone` and `Android`.
**Note:** Creating a new variable is ideal so that you don't overwrite original data.



In [7]:
device_to_code = {"iPhone": 1, "Android": 2}
df["device_code"]= df["device"].map(device_to_code)

In [6]:
df.head()

ID     label  sessions  drives  total_sessions  n_days_after_onboarding  \
0   0  retained       283     226      296.748273                     2276   
1   1  retained       133     107      326.896596                     1225   
2   2  retained       114      95      135.522926                     2651   
3   3  retained        49      40       67.589221                       15   
4   4  retained        84      68      168.247020                     1562   

   total_navigations_fav1  total_navigations_fav2  driven_km_drives  \
0                     208                       0       2628.845068   
1                      19                      64      13715.920550   
2                       0                       0       3059.148818   
3                     322                       7        913.591123   
4                     166                       5       3950.202008   

   duration_minutes_drives  activity_days  driving_days   device  device_code  
0              1985.775061             28            19  Android            2  
1              3160.472914             13            11   iPhone            1  
2              1610.735904             14             8  Android            2  
3               587.196542              7             3   iPhone            1  
4              1219.555924             27            18  Android            2

You are interested in the relationship between device type and the number of drives. One approach is to look at the average number of drives for each device type. Calculate these averages.

In [8]:
df.groupby("device")["drives"].mean()

device
Android    66.231838
iPhone     67.859078
Name: drives, dtype: float64

Based on the averages shown, it appears that drivers who use an iPhone device to interact with the application have a higher number of drives on average. However, this difference might arise from random sampling, rather than being a true difference in the number of drives. To assess whether the difference is statistically significant, you can conduct a hypothesis test.


### Hypothesis testing

Your goal is to conduct a two-sample t-test. Recall the steps for conducting a hypothesis test:


1.   State the null hypothesis and the alternative hypothesis
2.   Choose a signficance level
3.   Find the p-value
4.   Reject or fail to reject the null hypothesis

**Note:** This is a t-test for two independent samples. This is the appropriate test since the two groups are independent (Android users vs. iPhone users).

Recall the difference between the null hypothesis ($H_0$) and the alternative hypothesis ($H_A$).

**Question:** What are your hypotheses for this data project?

> $H_0$: mean drives are the same for iPhone and Android users

> $H_a$: mean drives are different depending on the OS

Next, choose 5% as the significance level and proceed with a two-sample t-test and perform the test.

In [10]:
# 1. Isolate the `drives` column for iPhone users.
iphone_drives = df[df["device_code"] == device_to_code["iPhone"]]["drives"]

# 2. Isolate the `drives` column for Android users.
android_drives = df[df["device_code"] == device_to_code["Android"]]["drives"]

# 3. Perform the t-test
stats.ttest_ind(a=iphone_drives, b=android_drives, equal_var=False)

TtestResult(statistic=1.463523206885235, pvalue=0.143351972680206, df=11345.066049381952)

**Question:** Based on the p-value you got above, do you reject or fail to reject the null hypothesis?

> P-value is 14%, which is more then 5%, so we **fail to reject** the hypothesis. This means that there is NO statistical significance in the difference of the mean rides between android and iphone users

### Communicate insights with stakeholders

Now that you've completed your hypothesis test, the next step is to share your findings with the Waze leadership team. Consider the following question as you prepare to write your executive summary:

* What business insight(s) can you draw from the result of your hypothesis test?

>
> * On average, both Android and iPhone users have the same number of drives (Android - \~66 drives last month, iPhone  \~68)
> * Hypothesis testing revealed that there is no statistical significance in the difference of the number of drives between these groups, so the OS used does not inlfuence how many drives a user  makes

